pipeline/scripts/13-excelExport.R

```diff
@@ -137,9 +144,44 @@ if (z_score == 1) {
   }
   colnames(outlist)[startcol:ncol(outlist)] <- cnames.z
   
+  # Extra output: calculate robust scaler (Zscores minus outliers in Controls)
+  # calculate mean and sd for Control group without outliers
+  outlist.noZ$avg.ctrls <- 0
+  outlist.noZ$sd.ctrls  <- 0
+  
+  robustscaler <- function(x, perc=5, control_col_ids) {
+    nr_toremove <- ceiling(length(control_col_ids)*perc/100)
+    sorted_x <- sort(as.numeric(x))
+    trimmed_x <- sorted_x[(nr_toremove+1):(length(sorted_x)-nr_toremove)]
+    return(trimmed_x)
+  }
+  
+  # only calculate robust Z-scores if there are enough Controls
+  if (length(patient_col_ids) > 10) {
+    for (metabolite_index in 1:nrow(outlist)) {
+      outlist.noZ$avg.ctrls[metabolite_index] <- mean(robustscaler(outlist.noZ[metabolite_index, control_col_ids], perc, control_col_ids))
+      outlist.noZ$sd.ctrls[metabolite_index]  <-   sd(robustscaler(outlist.noZ[metabolite_index, control_col_ids], perc, control_col_ids))
```

Comment:
Wat is de waarde van perc in dit geval? Geeft dit geen fout want perc bestaat niet denk ik.